### PR TITLE
update to mbed os 5.4.1 and use native file system

### PR DESCRIPTION
**Pull request by soleilplanet**
change list:
 - remove SDFileSystem.lib
 - add SDDriver.lib
 - update mbed-os.lib to 5.4.1
 - modify SPI test code to use mbed OS native file system

### CODE REVIEW

File: TESTS/API/SPI/SPI.cpp [hunk 1]
```diff
@@ -58,28 +59,59 @@ void test_object()
...
+	error = fs.mount(&sd);
+
     FILE *File = fopen("/sd/test_sd_w.txt", "w");   // open file
     TEST_ASSERT_MESSAGE(File != NULL,"SD Card is not present. Please insert an SD Card.");
     TEST_ASSERT_MESSAGE(fprintf(File, SD_TEST_STRING) > 0,"Writing file to sd card failed");             // write data
     // TODO: test that the file was written correctly
     fclose(File);                              // close file on SD
     TEST_ASSERT(true);
+
+	error = fs.unmount();
```
Comment (0xc0170) [inline TESTS/API/SPI/SPI.cpp]: can you please fix the formatting (using tabs on some lines?) line 98, line 107, line 87. Please check entire patch

File: TESTS/API/SPI/SPI.cpp [hunk 2]
```diff
@@ -58,28 +59,59 @@ void test_object()
...
+
+	sd.deinit();
 }
 
 // Test SD Card Write
 void test_sd_w()
 {
-    SDFileSystem sd(MBED_CONF_APP_SPI_MOSI, MBED_CONF_APP_SPI_MISO, MBED_CONF_APP_SPI_CLK, MBED_CONF_APP_SPI_CS,"sd");
+	int error = 0; 
+	SDBlockDevice sd(MBED_CONF_APP_SPI_MOSI, MBED_CONF_APP_SPI_MISO, MBED_CONF_APP_SPI_CLK, MBED_CONF_APP_SPI_CS);//alvin add
```
Comment (0xc0170) [inline TESTS/API/SPI/SPI.cpp]: ``//alvin add`` what does this comment mean?

File: TESTS/API/SPI/SPI.cpp [hunk 3]
```diff
@@ -58,28 +59,59 @@ void test_object()
...
 {
-    SDFileSystem sd(MBED_CONF_APP_SPI_MOSI, MBED_CONF_APP_SPI_MISO, MBED_CONF_APP_SPI_CLK, MBED_CONF_APP_SPI_CS,"sd");
-    FILE *File = fopen("/sd/card-present.txt", "w");   // open file
+    int error = 0; 
+	SDBlockDevice sd(MBED_CONF_APP_SPI_MOSI, MBED_CONF_APP_SPI_MISO, MBED_CONF_APP_SPI_CLK, MBED_CONF_APP_SPI_CS);
+	FATFileSystem fs("sd");
+	sd.init();
+
+	error = fs.mount(&sd);
+	TEST_ASSERT_MESSAGE(error==0,"SD file system mount failed.");
```
Comment (0xc0170) [inline TESTS/API/SPI/SPI.cpp]: spaces around ``error == 0``

File: TESTS/API/SPI/SPI.cpp [hunk 4]
```diff
@@ -58,28 +59,59 @@ void test_object()
...
+	
+    FILE *File = fopen("/sd/card-present.txt", "w+");   // open file
```
Comment (0xc0170) [inline TESTS/API/SPI/SPI.cpp]: the comment is redundant in this case, its clear that fopen is opening the file

**Comment by 0xc0170** [comment]
@soleilplanet Thanks for this patch ! There are just small nits above I left. What I would like to have is the test results for this patch. I can help to test this patch with some platforms. Please share what has been tested (paste here test results).

**Comment by soleilplanet** [comment]
Hi, I've made the changes and here is the test result:
[cits_log.txt](https://github.com/ARMmbed/ci-test-shield/files/850043/cits_log.txt)
Thanks!

**Comment by 0xc0170** [comment]
Thanks for the update. The log - tested with K64F. I'll run some test on various boards, will report back

We always should at least 3 different targets (different vendors).

**Comment by soleilplanet** [comment]
Hi, here's the log for STM Nucleo_F401RE:
[cits_log_nucleo.txt](https://github.com/ARMmbed/ci-test-shield/files/850101/cits_log_nucleo.txt)
I don't have other boards in hand, sorry for the inconvenience and thanks for helping!

**Comment by 0xc0170** [comment]
Tested with nuvoton NUC472, no regression (2 errors, same on master). NRF51DK, no regression. 

LGTM

**Comment by BlackstoneEngineering** [comment]
Verified on ST 429ZI board and K64F, Looks good to me, LGTM

**Pull request merged**